src/lib/datepicker/datepicker-input.ts

```diff
@@ -92,198 +101,35 @@ export class MatDatepickerInputEvent<D> {
   },
   exportAs: 'matDatepickerInput',
 })
-export class MatDatepickerInput<D> implements AfterContentInit, ControlValueAccessor, OnDestroy,
-    Validator {
+export class MatDatepickerInput<D> extends CdkDatepickerInput<D> implements AfterContentInit,
+    ControlValueAccessor, OnDestroy, Validator {
+  /** Prefix for form control validator properties. */
+  protected _formControlValidatorPrefix = 'mat';
+
   /** The datepicker that this input is associated with. */
   @Input()
   set matDatepicker(value: MatDatepicker<D>) {
-    this.registerDatepicker(value);
+    this.datepicker = value;
```

Comment:
I _think_ we can just rename these in the `inputs` list above

Reply:
Renaming in the `inputs` above didn't work, maybe setter/getter problem? We discussed this and decided to leave this as is.